Add vmware profile

Neo00001 opened the pull request.

Neo00001 commented:
`whitelist-usr-share-common.inc,whitelist-var-common.inc,whitelist-runuser-common.inc` 
- didn't feel the need. nothing seems broken regarding normal functionality. btw, `whitelist-runuser-common.inc` is not available under firejail version 0.9.62.

`nonewprivs,noroot,,protocol,seccomp,private-bin`
- all of these make vmware dysfunctional. vmware may launch but can't run any guest os.

`private-bin` 
- despite allowing all of vmware related bin files vmware still doesn't run. if you may suggest some other files I would like to check it further.

`private-cache` 
- drag & drop from guest to host won't work

`private-tmp` 
- it may work, I have to check it though. there is a folder namely VMwareDnD (in /tmp) which keeps linked folders of cache's drag_and_drop folder's contents.

`tracelog`
- it works. but is it needed?

rusty-snake commented:
> didn't feel the need. nothing seems broken regarding normal functionality

Adding them is more secure, then are only whitelisted files/dirs allowed, w/o all are allowed.
wusc may need extras (e.g. `whitelist /usr/share/vm-ware`)

> all of these make vmware dysfunctional. vmware may launch but can't run any guest os.

:+1: vmware is a suid, right?
- `protocol` which what protocols have tested? `protocol unix,inet,inet6,netlink,packet` is the most permissive one.
- `seccomp` you can watch your syslog (`journalctl --follow`) to know which syscall are blocked

> despite allowing all of vmware related bin files vmware still doesn't run. if you may suggest some other files I would like to check it further.

~You can use `firejail --build vmware`~ You can not since nonewprivs breaks.

> it works. but is it needed?

It logs blacklist violation which are usually a indicator for issues. In general it is added to all profiles where it is supported.

> whitelist-runuser-common.inc is not available under firejail version 0.9.62.

If you still use X11, you can copy it.

rusty-snake commented on etc/profile-m-z/vmware.profile:
```diff
@@ -0,0 +1,36 @@
+#disable-mnt
+private-etc private-etc alsa,asound.conf,ca-certificates,conf.d,dconf,fonts,gtk-2.0,gtk-3.0,hostname,hosts,ld.so.cache,localtime,machine-id,pulse,resolv.conf,ssl,vmware,vmware-installer,vmware-vix,X11
+
+dbus-* none
```
```suggestion
dbus-user none
dbus-system none
```

rusty-snake commented on etc/profile-m-z/vmware.profile:
```diff
@@ -0,0 +1,36 @@
+caps.keep sys_nice
+ipc-namespace
+netfilter
+# dbus-user none
+# dbus-system none
```
```suggestion


```

rusty-snake commented on etc/profile-m-z/vmware.profile:
```diff
@@ -0,0 +1,36 @@
+mkdir ${HOME}/.cache/vmware
+whitelist ${HOME}/.vmware
+whitelist ${HOME}/.cache/vmware
+
```
```suggestion
```

rusty-snake commented on etc/profile-m-z/vmware.profile:
```diff
@@ -0,0 +1,36 @@
+# Firejail profile for vmware
+# This file is overwritten after every install/update
+# Description: The industry standard for running multiple operating systems as virtual machines on a single Linux PC.
+include vmware.local
```
```suggestion
# Firejail profile for vmware
# Description: The industry standard for running multiple operating systems as virtual machines on a single Linux PC.
# This file is overwritten after every install/update
# Persistent local customizations
include vmware.local
# Persistent global definitions
include globals.local
```

rusty-snake commented on etc/profile-m-z/vmware.profile:
```diff
@@ -0,0 +1,36 @@
+notv
+shell none
+
+#disable-mnt
```
Breaks it something?

Neo00001 replied:
I haven't noticed anything broken. i have connected pendrive & an android device also. everything works as it should.

Neo00001 commented:
`private-tmp` is breaking drag & drop from guset to host

rusty-snake commented on etc/profile-m-z/vmware.profile:
```diff
@@ -0,0 +1,41 @@
+caps.keep sys_nice
+ipc-namespace
+netfilter
+
+
```
GitHub suggestions are still not perfect. Remove the blank lines.

Neo00001 replied:
done

Neo00001 commented:
`


> * `protocol` which what protocols have tested? `protocol unix,inet,inet6,netlink,packet` is the most permissive one.
> 
> * `seccomp` you can watch your syslog (`journalctl --follow`) to know which syscall are blocked


`protocol unix,inet,inet6,netlink,packet`
- doesn't allow vmware to run any guest os. 
& regarding seccomp, the syscall was iopl. & `seccomp !iopl` doesn't work

rusty-snake commented:
It maybe more then one, if `seccomp !foo,!bar` breaks without any logs, you can leave it out.

Neo00001 commented:
> It maybe more then one, if `seccomp !foo,!bar` breaks without any logs, you can leave it out.

`seccomp !iopl` breaks without logs.

Neo00001 commented:
should I add `nou2f`?

Neo00001 commented:
>     * private-dev

it's breaking vmware's functionality.vmware launches but can't run any guest os.

Neo00001 commented:
> Do not delete telegram.profile. You probably need to rebase -i and force push.

[should it be removed](https://github.com/netblue30/firejail/pull/3529)

rusty-snake commented:
:question:

Neo00001 commented:
> question

Sorry,I misunderstood your suggestion.

Neo00001 commented:
```whitelist-runuser-common.inc```
- It's breaking vmware's functionality.vmware launches but can't run any guest os.

Neo00001 commented:
> `whitelist-runuser-common.inc`
> 
>     * It's breaking vmware's functionality.vmware launches but can't run any guest os.

it is working now. I just had to run vmware without firejail once thereafter it worked again with firejail.

rusty-snake commented on etc/profile-m-z/vmware.profile:
```diff
@@ -22,6 +22,7 @@ mkdir ${HOME}/.cache/vmware
 whitelist ${HOME}/.vmware
 whitelist ${HOME}/.cache/vmware
 include whitelist-common.inc
+whitelist-runuser-common.inc
```
Missing include.

Is the socket in /run/user/$UID in a dir?

Neo00001 replied:
which socket? vmware doesn't use any socket? it uses systemd services though. 

btw, rather than allowing specific syscalls? can specific system call sets be applied?

rusty-snake replied:
>  rather than allowing specific syscalls? can specific system call sets be applied?

?? you mean groups?

Neo00001 replied:
yea, the pre-defined sets of system calls.

rusty-snake replied:
> which socket? vmware doesn't use any socket? it uses systemd services though.

what is the output of `ls /run/user/$UID -R`?

Neo00001 replied:
[ls /run/user/$UID -R](https://gist.github.com/Neo00001/f9a79b8ff93afb950d1d2ceadbfd9c1d)

rusty-snake replied:
>  I just had to run vmware without firejail once thereafter it worked again with firejail.

There is no obvious vmware socket, but it maybe need access to a systemd socket or something else.

Neo00001 replied:
keep `whitelist-runuser-common.inc` for now. if this problem persists, then it can be omitted. btw,at that point I also enabled `seccomp.block-secondary`, could that be the problem?  after commenting out `seccomp.block-secondary`, vmware misbehaved until I launched it without firejail.

rusty-snake replied:
If `seccomp` fails, `seccomp.block-secondary` will fail too.

Neo00001 replied:
I thought `seccomp.block-secondary` will only disable all non-native ABIs for system calls.Now,I do realize my wrong inference.If `seccomp` is still enabled for native syscalls then for sure it's useless in this context.

rusty-snake replied:
> it's useless

No, but if `seccomp !opl` fails w/o a log for a blocked syscall, vmware is one of the few prgorams which don't like it if they run under seccomp.

Neo00001 replied:
by useless, I meant to say that it can't be used for vmware profile. I didn't say seccomp as a feature is useless.

Neo00001 commented:
> I just had to run vmware without firejail once thereafter it worked again with firejail.

It's happening due to POSIX capabilities. It seems `chown` capability is needed for normal functionalities of vmware.

rusty-snake commented:
@Neo00001 remove the merge conflict (e.g. rebase and remove the telegram commits).

rusty-snake commented on etc/profile-m-z/telegram.profile:
```diff
@@ -25,5 +25,21 @@ seccomp
 
 disable-mnt
 private-cache
+<<<<<<< HEAD
+<<<<<<< HEAD
+<<<<<<< HEAD
+<<<<<<< HEAD
+private-etc ca-certificates,crypto-policies,fonts,ld.so.cache,localtime,machine-id,pki,pulse,resolv.conf,ssl
+=======
+private-etc fonts,resolv.conf,ld.so.cache,localtime,ca-certificates,ssl,pki,pulse
+>>>>>>> Create telegram.profile
+=======
+private-etc fonts,resolv.conf,ld.so.cache,localtime,ca-certificates,ssl,pki,pulse
+>>>>>>> Revert "Delete telegram.profile"
+=======
+private-etc fonts,resolv.conf,ld.so.cache,localtime,ca-certificates,ssl,pki,pulse
+>>>>>>> Revert "Delete telegram.profile"
+=======
+private-etc ca-certificates,crypto-policies,fonts,ld.so.cache,localtime,machine-id,pki,pulse,resolv.conf,ssl
+>>>>>>> parent of 17517516... Create telegram.profile
```
^

rusty-snake replied:
@Neo00001 commenting does not help.

Neo00001 replied:
It seems I have messed up this one. Can you help me out? btw,I'm using github-desktop

rusty-snake replied:
I have no experience with github-desktop. I found [this](https://stackoverflow.com/questions/37492908/is-there-a-way-to-make-github-desktop-rebase-a-branch-against-master). IDK if this is helpfull.

----

On the commandline it would be.
```
git checkout your_branch
git rebase -i master # where master should be at same HEAD as netblue30:master
```

https://docs.github.com/en/github/using-git/about-git-rebase
https://docs.github.com/en/github/using-git/using-git-rebase-on-the-command-line

Keep in mind that rebasing it a create thing to totally mess up a history (i.e. have a backup or first try in a test copy).

Neo00001 replied:
`git rebase -i master` - it's showing 
```
etc/profile-m-z/telegram.profile: needs merge
error: cannot rebase: You have unstaged changes.
error: additionally, your index contains uncommitted changes.
error: Please commit or stash them.
```

what shoud I do?

Neo00001 replied:
Can this be deleted? So that I can re-open a new PR.

rusty-snake commented:
Reseted, rebased and squashed that for you. Merged in https://github.com/netblue30/firejail/commit/6fa68d11ae3f8764e5acca9bfc09fcd2352619d9.

PS: You should delete your fork and fork again.

Neo00001 commented:
> Reseted, rebased and squashed that for you. Merged in [6fa68d1](https://github.com/netblue30/firejail/commit/6fa68d11ae3f8764e5acca9bfc09fcd2352619d9).
> 
> PS: You should delete your fork and fork again.

Thanks. 

& Sorry for all of these.
